src/Pos/Block/Arbitrary.hs

```diff
@@ -167,12 +157,13 @@ instance Arbitrary (SscPayloadDependsOnSlot ssc) =>
         return $ T.MainBody txPayload mpcData dlgPayload mpcUpload
 
 instance Arbitrary (SscPayload ssc) => Arbitrary (T.Body (T.MainBlockchain ssc)) where
-    arbitrary = makeSmall $ do
-        txPayload   <- arbitrary
-        mpcData     <- arbitrary
-        mpcProxySKs <- arbitrary
-        mpcUpload   <- arbitrary
-        return $ T.MainBody txPayload mpcData mpcProxySKs mpcUpload
+    arbitrary = makeSmall genericArbitrary
+    shrink mb = [ T.MainBody txp sscp dlgp updp
```

Comment:
No way, I don't think that the list monad (or list comprehension) is the right approach to writing shrink. This will try to shrink each field separately, creating a blowing up the space of shrinking choices. 

I refer you to the Haddock documentation of the `shrink` method where this issue is mentioned:
![image](https://user-images.githubusercontent.com/4061728/27662091-2845545c-5c65-11e7-94d3-41e9881fc1a9.png)